packages/block-editor/src/store/reducer.js

```diff
@@ -2131,6 +2140,524 @@ const combinedReducers = combineReducers( {
+ * @param {string} clientId  The client ID of the block to search the parents of.
+ * @param {Array}  clientIds The client IDs of the blocks to check.
+ *
+ * @return {string|undefined} The client ID of the parent block if found, undefined otherwise.
+ */
+function findParentInClientIdsList( state, clientId, clientIds ) {
+	let parent = state.blocks.parents.get( clientId );
+	while ( parent ) {
+		if ( clientIds.includes( parent ) ) {
+			return parent;
+		}
+		parent = state.blocks.parents.get( parent );
+	}
+}
+
+/**
+ * Checks if a block has any bindings in its metadata attributes.
+ *
+ * @param {Object} block The block object to check for bindings.
+ * @return {boolean}    True if the block has bindings, false otherwise.
+ */
+function hasBindings( block ) {
+	return (
+		block?.attributes?.metadata?.bindings &&
+		Object.keys( block?.attributes?.metadata?.bindings ).length
+	);
+}
+
+/**
+ * Determines the editing mode for a given block based on various factors.
+ *
+ * This function evaluates the editing mode of a block considering the following:
+ * - Whether the editor is in zoomed out or navigation mode.
+ * - The block's relationship to sections and synced patterns.
+ * - The presence of bindings in the block's attributes.
+ *
+ * @param {Object}  state                          The current state object.
+ * @param {string}  clientId                       The client ID of the block to evaluate.
+ * @param {Object}  options                        Additional options for determining the editing mode.
+ * @param {string}  options.sectionRootClientId    The client ID of the section root block.
+ * @param {Array}   options.sectionClientIds       An array of client IDs for section blocks.
+ * @param {Array}   options.syncedPatternClientIds An array of client IDs for synced pattern blocks.
+ * @param {boolean} options.isZoomedOut            Whether the editor is in zoomed out mode.
+ * @param {boolean} options.isNavMode              Whether the editor is in navigation mode.
+ *
+ * @return {string|undefined} The determined editing mode for the block: 'contentOnly', 'disabled', or undefined.
+ */
+function getDerivedBlockEditingModeForBlock(
+	state,
+	clientId,
+	{
+		sectionRootClientId,
+		sectionClientIds,
+		syncedPatternClientIds,
+		isZoomedOut,
+		isNavMode,
+	}
+) {
+	if ( isZoomedOut || isNavMode ) {
+		// If the root block is the section root set its editing mode to contentOnly.
+		if ( clientId === sectionRootClientId ) {
+			return 'contentOnly';
+		}
+
+		// There are no sections, so everything else is disabled.
+		if ( ! sectionClientIds?.length ) {
+			return 'disabled';
+		}
+
+		if ( sectionClientIds.includes( clientId ) ) {
+			return 'contentOnly';
+		}
+
+		// If zoomed out, all blocks that aren't sections or the section root are
+		// disabled.
+		// If the tree root is not in a section, set its editing mode to disabled.
+		if (
+			isZoomedOut ||
+			! findParentInClientIdsList( state, clientId, sectionClientIds )
+		) {
```

Comment:
This means if zoom out always set all blocks not sections diabled, if nav mode which unintuitively now is actually wrrite mode, then set to disabled all blocks that don't have section parents. It's pretty intricate.